### PR TITLE
(TFECO-7339) feat: Support context references within orchestrate blocks in deployment configuration

### PR DESCRIPTION
**Pull request by ansgarm**
depends on https://github.com/hashicorp/terraform-schema/pull/398

context

### CODE REVIEW

File: internal/features/stacks/jobs/builtin_references.go [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+	}
+
+	// The names of the existing components in the stack
+	// We use this to offer completions for the component_changes map
+	componentNames := make([]string, 0)
+	for name := range record.Meta.Components {
+		componentNames = append(componentNames, name)
+	}
+
+	for _, rng := range ranges {
```
Comment (jpogran) [inline internal/features/stacks/jobs/builtin_references.go:52]: Do we need to store the ranges again when they're already in the shape we need here?

File: internal/features/stacks/jobs/builtin_references.go [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+		// create the static base targets (like context.operation, context.success, etc.)
+		targets = append(targets, baseTargets(rng)...)
+		// create the static plan targets (like context.plan.mode, context.plan.applyable, etc.)
+		targets = append(targets, staticPlanTargets(rng)...)
+
+		// targets for each component for the component_changes map (like context.plan.component_changes["vpc"].total)
+		for _, name := range componentNames {
```
Comment (jpogran) [inline internal/features/stacks/jobs/builtin_references.go:59]: Same question as above, not sure they need to be new allocations

File: internal/features/stacks/jobs/builtin_references.go (hunk 1 above)
Comment (ansgarm) [reply internal/features/stacks/jobs/builtin_references.go]: Yeah, makes sense! I was unsure whether I would need this multiple time but ended up with just two loops over them 👍

**Review by jpogran — APPROVED** [review]
:shipit:

**Pull request merged**